Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -385,8 +385,11 @@ class Gx2Fitter {
                     const Logger& /*logger*/) const {
       assert(result.fittedStates && "No MultiTrajectory set");
 
-      if (state.navigation.navigationBreak) {
-        ACTS_INFO("Actor: finish: state.navigation.navigationBreak");
+      if (result.measurementStates == inputMeasurements->size() ||
+          state.navigation.navigationBreak) {
+        ACTS_INFO(
+            "Actor: finish: All measurements are found or navigationBreak is "
+            "received.");
```

Comment:
I wonder if it would be good to know which one of both triggered the finish

Reply:
I also thought about it. Let's do it. I added some comments around to make it less cluttered.